service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ObservationStatementMapperTest.java

```diff
@@ -0,0 +1,129 @@
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.time.ZoneOffset;
+import java.util.TimeZone;
+
+import uk.nhs.adaptors.gp2gp.common.service.FhirParseService;
+import uk.nhs.adaptors.gp2gp.common.service.RandomIdGeneratorService;
+import uk.nhs.adaptors.gp2gp.ehr.exception.EhrMapperException;
+import uk.nhs.adaptors.gp2gp.utils.ResourceTestFileUtils;
+
+import org.hl7.fhir.dstu3.model.Observation;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class ObservationStatementMapperTest {
+    private static final String TEST_ID = "394559384658936";
+    private static final String TEST_FILE_DIRECTORY = "/ehr/mapper/observation/";
+    private static final String INPUT_JSON_WITH_EFFECTIVE_DATE_TIME = TEST_FILE_DIRECTORY + "example-observation-resource-1.json";
+    private static final String INPUT_JSON_WITH_NULL_DATE_TIME = TEST_FILE_DIRECTORY + "example-observation-resource-2.json";
+    private static final String INPUT_JSON_WITH_EFFECTIVE_PERIOD = TEST_FILE_DIRECTORY + "example-observation-resource-3.json";
+    private static final String INPUT_JSON_WITH_ISSUED_ONLY = TEST_FILE_DIRECTORY + "example-observation-resource-4.json";
+    private static final String INPUT_JSON_WITH_NO_DATES = TEST_FILE_DIRECTORY + "example-observation-resource-5.json";
+    private static final String OUTPUT_XML_USES_EFFECTIVE_DATE_TIME = TEST_FILE_DIRECTORY + "expected-output-observation-statement-1.xml";
+    private static final String OUTPUT_XML_USES_UNK_DATE_TIME = TEST_FILE_DIRECTORY + "expected-output-observation-statement-2.xml";
+    private static final String OUTPUT_XML_USES_NESTED_COMPONENT = TEST_FILE_DIRECTORY + "expected-output-observation-statement-3.xml";
+    private static final String OUTPUT_XML_USES_EFFECTIVE_PERIOD = TEST_FILE_DIRECTORY + "expected-output-observation-statement-4.xml";
+
+    private CharSequence expectedOutputMessage;
+    private ObservationStatementMapper observationStatementMapper;
+
+    @Mock
+    private RandomIdGeneratorService randomIdGeneratorService;
+
+    @BeforeAll
+    public static void initialize() {
+        TimeZone.setDefault(TimeZone.getTimeZone(ZoneOffset.UTC));
+    }
+
+    @AfterAll
+    public static void deinitialize() {
+        TimeZone.setDefault(null);
+    }
+
+    @BeforeEach
+    public void setUp() {
+        when(randomIdGeneratorService.createNewId()).thenReturn(TEST_ID);
+        observationStatementMapper = new ObservationStatementMapper(randomIdGeneratorService);
+    }
+
+    @Test
+    public void When_MappingParsedObservationJsonWithEffectiveDatetime_Expect_ObservationStatementXmlOutput() throws IOException {
```

Comment:
This test can be parametrised. Please take a look at "EhrExtractRequestHandlerTest" for a reference.